File: scratch/fix_button_colors.py
import os
import re

directory = 'c:/xampp/htdocs/sistem-pkl-v13/resources/views'

for root, dirs, files in os.walk(directory):
    for file in files:
        if file.endswith('.blade.php'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
                
            # Replace text-slate-900 dark:text-white with text-white if bg-blue-600 is present on the same line
            # Wait, regular expressions are better.
            
            lines = content.split('\n')
            changed = False
            for i, line in enumerate(lines):
                if 'bg-blue-600' in line and 'text-slate-900 dark:text-white' in line:
                    lines[i] = line.replace('text-slate-900 dark:text-white', 'text-white')
                    changed = True
                elif 'bg-indigo-600' in line and 'text-slate-900 dark:text-white' in line:
                    lines[i] = line.replace('text-slate-900 dark:text-white', 'text-white')
                    changed = True
                    
            if changed:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write('\n'.join(lines))
                print(f"Updated {filepath}")

File: scratch/fix_button_colors.php
<?php
$directory = 'c:/xampp/htdocs/sistem-pkl-v13/resources/views';

$iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($directory));
$regex = new RegexIterator($iterator, '/^.+\.blade\.php$/i', RecursiveRegexIterator::GET_MATCH);

foreach ($regex as $file => $object) {
    $content = file_get_contents($file);
    $lines = explode("\n", $content);
    $changed = false;
    
    foreach ($lines as &$line) {
        if (strpos($line, 'bg-blue-600') !== false && strpos($line, 'text-slate-900 dark:text-white') !== false) {
            $line = str_replace('text-slate-900 dark:text-white', 'text-white', $line);
            $changed = true;
        }
    }
    
    if ($changed) {
        file_put_contents($file, implode("\n", $lines));
        echo "Updated: $file\n";
    }
}
echo "Done.\n";
